src/models.js

```diff
@@ -306,6 +258,70 @@ module.exports = ({ cooler, isPublic }) => {
         })) || "";
       return raw;
     },
+    // This needs to run in the background but also needs to be aborted
+    // in index.js when the server closes. There's also an interval that
+    // needs to be cleared. TODO: Ensure that there's never more than
+    // one interval running at a time.
+    _startNameWarmup() {
+      const abortable = pullAbortable();
+      let intervals = [];
```

Comment:
I think the only thing I don't understand about this patch is this change, to have a *list* of intervals...

Reply:
Potential problem: I didn't see any code that ensures that we only have one interval running at a time. I _think_ that the 'sync' events happen every time we're up-to-date, which gives me the impression that we might have multiple intervals active at once? I don't know, didn't have time to test but also *really* didn't want to miss any of these things.

Happy to merge an improvement removing the list if we can be sure there's only one interval active at a time.

Reply:
okay so after looking at this a bit harder.... I think I see why you made this into a list of intervals. you're expecting `msg.sync` to be `true` more than once?
If that's the case, then that should have already been the case before, no? 

What I don't see in your code though is something that would ever remove an interval id from the list again. 
Shouldn't the code that's triggered by the interval remove it from the list again?

Reply:
> If that's the case, then that should have already been the case before, no?

Yep, exactly. Previously this interval wasn't being cleared, which is one reason why the tests hung forever -- this clears the intervals, but since I'm not sure how many there are then I'm double-checking to make sure there can't be more than one.

(This could also be accomplished with an `if`.)

> What I don't see in your code though is something that would ever remove an interval id from the list again.

Like this, or do you mean something else?

https://github.com/fraction/oasis/pull/465/files#diff-a19725de73a858e12dc293aef7c29e8d9d107497643f2c65175ab0064d804665R321

Reply:
That clears the interval, yes. But the integer (?) referencing this interval will still be held in the `intervals` variable, no?

Reply:
(sorry if I'm wasting your time. I haven't used these before, not really, and I'm just going by what I find in MDN)

Reply:
That's right! But this only happens when the Oasis closes so we don't need to worry about removing items from memory. If we were clearing items every few minutes or so, you're absolutely correct that this would cause a memory leak. I'd be happy to merge a PR improving this to do it a different way, I mostly just want to fix up the tests.

Reply:
No, just checking that I understand what's happening. Clearing up a few bytes just before exit seems... non-prioritary :P